provisioning_templates/provision/kickstart_default.erb

```diff
@@ -91,14 +80,87 @@ selinux --<%= host_param('selinux-mode') || host_param('selinux') || 'enforcing'
 keyboard <%= host_param('keyboard') || 'us' %>
 skipx
 
-<% subnet = @host.subnet -%>
-<% if subnet.respond_to?(:dhcp_boot_mode?) -%>
-<% dhcp = subnet.dhcp_boot_mode? && !@static -%>
-<% else -%>
-<% dhcp = !@static -%>
-<% end -%>
+<%
+  network_options = []
+  nameservers = []
+  mtus = []
+  subnet4 = iface.subnet
+  subnet6 = iface.subnet6
+
+  # device and hostname
+  network_options.push("--device=#{iface.mac || iface.identifier}")
+  network_options.push("--hostname #{@host.name}")
+
+  # single stack
+  if subnet4 && !subnet6
+    network_options.push("--noipv6")
+  elsif !subnet4 && subnet6
+    network_options.push("--noipv4")
+  end
+
+  # IPv4
+  if (subnet4 && !subnet4.dhcp_boot_mode?) || @static
+    network_options.push("--bootproto static")
+    network_options.push("--ip=#{iface.ip}")
+    network_options.push("--netmask=#{subnet4.mask}")
+    network_options.push("--gateway=#{subnet4.gateway}")
+  elsif subnet4 && subnet4.dhcp_boot_mode?
+    network_options.push("--bootproto dhcp")
+  end
+  if subnet4
+    nameservers.push(subnet4.dns_servers)
+    mtus.push(subnet4.mtu) if subnet4.mtu.present?
+  end
+
+  # IPv6
+  if rhel_compatible && os_major >= 6
+    if (subnet6 && !subnet6.dhcp_boot_mode?) || @static6
+      network_options.push("--ipv6=#{iface.ip6}/#{subnet6.cidr}")
+      network_options.push("--ipv6gateway=#{subnet6.gateway}")
+    elsif subnet6 && subnet6.dhcp_boot_mode?
+      if host_param_true?('disable-slaac')
+        network_options.push("--ipv6 dhcp")
+      else
+        network_options.push("--ipv6 auto")
+      end
+    end
+    if subnet6
+      nameservers.push(subnet6.dns_servers)
+      mtus.push(subnet6.mtu) if subnet6.mtu.present?
+    end
+  end
+
+  # bond
+  if iface.bond? && rhel_compatible && os_major >= 6
+    network_options.push("--bondslaves=#{iface.attached_devices_identifiers}")
+    network_options.push("--bondopts=mode=#{iface.mode};#{iface.bond_options.tr(' ', ';')}")
+  end
 
-network --bootproto <%= dhcp ? 'dhcp' : "static --ip=#{@host.ip} --netmask=#{subnet.mask} --gateway=#{subnet.gateway} --nameserver=#{subnet.dns_servers.join(',')} --mtu=#{subnet.mtu.to_s}" %> --hostname <%= @host %> <%= network_options.join(' ') %>
+  # VLAN (only on physical is recognized)
+  if iface.virtual? && iface.tag.present? && iface.attached_to.present?
+    if rhel_compatible && os_major >= 6
+      network_options.push("--vlanid=#{iface.tag}")
+    elsif rhel_compatible && os_major >= 7
+      network_options.push("--interfacename=vlan#{iface.tag}")
+    end
+  end
+
+  # DNS
+  if nameservers.flatten.size > 0
+    network_options.push("--nameserver=#{nameservers.flatten.join(',')}")
+  else
+    network_options.push("--nodns")
+  end
+
+  # minimum MTU takes precedence for dual-stack setup
+  # min/max methods are not allowed: https://github.com/svenfuchs/safemode/pull/30
+  if mtus[1] && mtus[1] > mtus[0]
```

Comment:
I wonder if we should take a min() or max() approach. This implements max() but won't that lead to incompatibilities and MTU issues? Let's say IPv6 allows jumbo frames (MTU 9000) but IPv4 only standard (MTU 1500) it will cause a lot of fragmentation for IPv4. Not an expert in this so carefully consider this.

Reply:
Good point, this could create terrible pains and investigation. I am opting for explicit solution - if MTUs are different, then let's raise an exception.

Reply:
In IPv6 each route can have its own MTU. I'm not sure what the best approach is.

@jovandeginste @vstone AFAIK you've both done your fair share of networking. Care to weigh in here?

Reply:
No, provisioning interface can only have one IPv4 and one IPv6 network (which has MTU assigned in Foreman DB). Unless it does not differ, we don't care.

Reply:
@ekohl Never did different routes with different MTU on the same interface...